brave/brave-ios
# Fix #3177: Update pre-populated search engine list

This also replaces Yahoo with Ecosia on search onboarding screen.



## Summary of Changes


This pull request fixes #3177 

## Submitter Checklist:


## Test Plan:


1. New user
- Set region with Ecosia enabled, United States for example
- Fresh install the app
- Verify that Ecosia is one of search options showed during onboarding
- Verify that Yahoo is not listed as one of search options
- Select it, make sure search queries use this search engine
- Verify that correct search parameters are passed, you can refer to https://github.com/brave/brave-browser/wiki/Search-engine-integrations

2. Existing user
- Install previous app build, select some search engine
- Update to build with Ecosia, make sure eligible region(like US) is set
- Go to Brave Settings-> search engines
- Verify that Ecosia is on the list, select it
- Verify that correct search parameters are passed

3. Existing Yahoo user
- Install previous app build, select Yahoo as default search engine
- Update to build with Ecosia, make sure eligible region(like US) is set
- Verify that default search is still Yahoo


## Screenshots:

<img width="520" alt="Zrzut ekranu 2021-01-12 o 15 21 20" src="https://user-images.githubusercontent.com/6950387/104326311-e6ba1500-54e9-11eb-889b-fa111d7d7b90.png">

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
